app/assets/javascripts/discourse/controllers/bookmark.js.es6

```diff
@@ -0,0 +1,224 @@
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import discourseComputed from "discourse-common/utils/decorators";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { htmlSafe } from "@ember/template";
+import { ajax } from "discourse/lib/ajax";
+
+const START_OF_DAY_HOUR = 8;
+const REMINDER_TYPES = {
+  AT_DESKTOP: "at_desktop",
+  LATER_TODAY: "later_today",
+  NEXT_BUSINESS_DAY: "next_business_day",
+  TOMORROW: "tomorrow",
+  NEXT_WEEK: "next_week",
+  NEXT_MONTH: "next_month",
+  CUSTOM: "custom"
+};
+
+export default Controller.extend(ModalFunctionality, {
+  loading: false,
+  errorMessage: null,
+  name: null,
+  selectedReminderType: null,
+  closeWithoutSaving: false,
+  saveButtonClicked: false,
+  onCloseWithoutSaving: null,
+
+  onShow() {
+    this.setProperties({
+      errorMessage: null,
+      loading: true,
+      name: null,
+      selectedReminderType: null,
+      closeWithoutSaving: false,
+      saveButtonClicked: false
+    });
+
+    this.set("loading", false);
+  },
+
+  // we always want to save the bookmark unless the user specifically
+  // clicks the save or cancel button to mimic browser behaviour
+  onClose() {
+    if (!this.closeWithoutSaving && !this.saveButtonClicked) {
+      this.saveBookmark();
+    }
+    if (this.onCloseWithoutSaving && this.closeWithoutSaving) {
+      this.onCloseWithoutSaving();
+    }
+  },
+
+  @discourseComputed()
+  usingMobileDevice() {
+    return this.site.mobileView;
+  },
+
+  @discourseComputed()
+  reminderTypes() {
```

Comment:
this would work probably, I have a doubt:

``` 
reminderTypes: REMINDER_TYPES
```

or 

```
reminderTypes: computed(function() { return REMINDER_TYPES; })
```